autocompleter/base.py

```diff
@@ -630,15 +630,22 @@ def suggest(self, term, facets=[]):
                     # exact term matches don't bypass the requirement of having matching facet values.
                     # To achieve this, we append the previous intermediate result key (which at this point will
                     # contain all the facet matches) to the list of exact match keys and perform an intersection.
-                    keys.append(base_result_key)
-                    old_base_result_key = base_result_key
-                    base_result_key = RESULT_SET_BASE_NAME % str(uuid.uuid4())
-                    pipe.zinterstore(base_result_key, keys, aggregate='MIN')
-                    pipe.delete(old_base_result_key)
+                    facet_exact_match_key = RESULT_SET_BASE_NAME % str(uuid.uuid4())
+                    keys.append(faceted_base_result_key)
+                    pipe.zinterstore(facet_exact_match_key, keys, aggregate='MIN')
                 else:
                     pipe.zunionstore(base_result_key, keys, aggregate='MIN')
-                pipe.zrange(base_result_key, 0, MAX_RESULTS - 1)
+
+                if use_facets:
+                    pipe.zrange(facet_exact_match_key, 0, MAX_RESULTS - 1)
+                    pipe.delete(facet_exact_match_key)
+                else:
+                    pipe.zrange(base_result_key, 0, MAX_RESULTS - 1)
+
+            pipe.delete(base_term_result_key)
             pipe.delete(base_result_key)
```

Comment:
let's think about these deletes .... are they necessary at all
I would say yes if we store something in the key in each provider loop where we use it, if that makes sense